sdk/core/azure-core/src/main/java/com/azure/core/models/CloudEvent.java

```diff
@@ -0,0 +1,470 @@
+     */
+    @JsonAnyGetter
+    public Map<String, Object> getExtensionAttributes() {
+        return this.extensionAttributes;
+    }
+
+    /**
+     * Add/Overwrite a single extension attribute to the cloud event.
+     * @param name the name of the attribute. It must contains only alphanumeric characters and not be be any
+     *             CloudEvent reserved attribute names.
+     * @param value the value to associate with the name.
+     *
+     * @return the cloud event itself.
+     * @throws IllegalArgumentException if name format isn't correct.
+     */
+    @JsonAnySetter
+    public CloudEvent addExtensionAttribute(String name, Object value) {
+        if (Objects.isNull(name)) {
+            throw LOGGER.logExceptionAsError(new IllegalArgumentException("'name' cannot be null."));
+        }
+        if (Objects.isNull(value)) {
+            throw LOGGER.logExceptionAsError(new IllegalArgumentException("'value' cannot be null."));
+        }
+        if (!validateAttributeName(name)) {
+            throw LOGGER.logExceptionAsError(new IllegalArgumentException(
+                "'name' must have only small-case alphanumeric characters and not be one of the CloudEvent reserved "
+                    + "attribute names"));
+        }
+        if (this.extensionAttributes == null) {
+            this.extensionAttributes = new HashMap<>();
+        }
+        this.extensionAttributes.put(name.toLowerCase(Locale.ENGLISH), value);
+        return this;
+    }
+
+    /**
+     * Get the spec version. Users don't need to access it because it's always 1.0.
+     * Make it package level to test deserialization.
+     * @return The spec version.
+     */
+    String getSpecVersion() {
+        return this.specVersion;
+    }
+
+    /**
+     * Set the spec version. Users don't need to access it because it's always 1.0.
+     * Make it package level to test serialization.
+     * @return the cloud event itself.
+     */
+    CloudEvent setSpecVersion(String specVersion) {
+        this.specVersion = specVersion;
+        return this;
+    }
+
+    private static boolean validateAttributeName(String name) {
+        if (RESERVED_ATTRIBUTE_NAMES.contains(name)) {
+            return false;
+        }
+        for (int i = 0; i < name.length(); i++) {
+            char c = name.charAt(i);
+            if (!((c >= 'a' && c <= 'z') || (c >= '0' && c <= '9'))) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    static class JacksonSerializer implements JsonSerializer {
+        private final JacksonAdapter jacksonAdapter = new JacksonAdapter();
+
+        @Override
+        public <T> T deserialize(InputStream stream, TypeReference<T> typeReference) {
+            try {
+                return jacksonAdapter.deserialize(stream, typeReference.getJavaType(), SerializerEncoding.JSON);
+            } catch (IOException e) {
+                throw LOGGER.logExceptionAsError(new RuntimeException(e));
```

Comment:
I would use `UncheckedIOException` as the non-checked wrapper exception type as it more closely related to `IOException`